packages/components/organisms/f-takeawaypay-activation/src/components/TakeawaypayActivation.vue

```diff
@@ -55,12 +68,107 @@ export default {
         homeUrl: {
             type: String,
             required: true
+        },
+        authToken: {
+            type: String,
+            default: ''
         }
     },
     data () {
         return {
-            tenantConfigs
+            tenantConfigs,
+            consumerId: '',
+            consumerEmail: '',
+            activationState: ACTIVATION_STATE_NONE
         };
+    },
+
+    computed: {
+        activationStateComponent () {
+            if (this.activationState === ACTIVATION_STATE_AVAILABLE_LOGGED_IN) {
+                return this.activationLoggedIn;
+            }
+
+            if (this.activationState === ACTIVATION_STATE_AVAILABLE_NOT_LOGGED_IN) {
+                return this.activationNotLoggedIn;
+            }
+
+            if (this.activationState === ACTIVATION_STATE_SUCCEEDED) {
+                return this.activationSucceeded;
+            }
+            return this.activationFailed;
+        },
+
+        activationFailed () {
+            return {
+                name: 'activation-failed',
+                props: {
+                    'redirect-url': this.redirectUrl
```

Comment:
Here and below the prop keys can be passed like:
```suggestion
                    redirectUrl: this.redirectUrl
```